pkg/development/devcontext.go

```diff
@@ -102,6 +109,56 @@ func newDevContextWithDatastore(ctx context.Context, requestContext *devinterfac
 	}, nil, nil
 }
 
+// RunV1InMemoryService runs a V1 server in-memory on a buffconn over the given
+// development context and returns a client connection and a function to shutdown
+// the server. It is the responsibility of the caller to call the function to close
+// the server.
+func (dc *DevContext) RunV1InMemoryService() (*grpc.ClientConn, func(), error) {
+	buffer := 1024 * 1024
+	listener := bufconn.Listen(buffer)
+
+	s := grpc.NewServer(
+		grpc.ChainUnaryInterceptor(
+			datastoremw.UnaryServerInterceptor(dc.Datastore),
+			consistency.UnaryServerInterceptor(),
+		),
+		grpc.ChainStreamInterceptor(
+			datastoremw.StreamServerInterceptor(dc.Datastore),
+			consistency.StreamServerInterceptor(),
+		),
```

Comment:
Perhaps we could move this to `defaults.go` as a function like `RequiredMiddleware` that is then reused in `DefaultMiddleware`. These are the _required_ middleware for SpiceDB to actually properly work, but it's not explicit anywhere. In `defaults.go`, with recent changes introduced in https://github.com/authzed/spicedb/pull/1063, we tried to capture what are "optional middleware" and "required middleware" by marking a middleware as `internal`.

Reply:
I can't have this import any file that relies upon the other middlewares, otherwise it won't build under WASM. This was placed here explicitly to ensure we only need the bare minimum, while also not breaking the existing WASM build.

Reply:
let's then turn that into a method like `defaultWasmMiddleware` in this package? WDYT?

Reply:
Just for this one usage? Why?

Reply:
makes more explicit what middleware is needed to run a barebone SpiceDB in wasm?

It's also fine to leave it as is - my initial assumption was this was the required middleware for a normal spicedb deployment as well